pkg/test/component_sns_sqs.go

```diff
@@ -0,0 +1,131 @@
+package test
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/service/sns"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"log"
+	"net"
+	"strings"
+	"time"
+)
+
+type snsSqsConfig struct {
+	Debug   bool   `mapstructure:"debug"`
+	Host    string `mapstructure:"host"`
+	SnsPort int    `mapstructure:"sns_port"`
+	SqsPort int    `mapstructure:"sqs_port"`
+}
+
+var snsSqsConfigs map[string]*snsSqsConfig
+
+var localstackClients = simpleCache{}
+
+func init() {
+	snsSqsConfigs = map[string]*snsSqsConfig{}
+	localstackClients = simpleCache{}
+}
+
+// do we really need this ?
+func onDestroy() {
+	snsSqsConfigs = map[string]*snsSqsConfig{}
+	localstackClients = simpleCache{}
+}
+
+func ProvideSnsClient(name string) *sns.SNS {
+	return localstackClients.New("sns-"+name, func() interface{} {
+		sess, err := getSession(snsSqsConfigs[name].Host, snsSqsConfigs[name].SnsPort)
+
+		if err != nil {
+			logErr(err, "could not create sns client: %s")
+		}
+
+		return sns.New(sess)
+	}).(*sns.SNS)
+}
+
+func ProvideSqsClient(name string) *sqs.SQS {
+	return localstackClients.New("sqs-"+name, func() interface{} {
+		sess, err := getSession(snsSqsConfigs[name].Host, snsSqsConfigs[name].SqsPort)
+
+		if err != nil {
+			logErr(err, "could not create sqs client: %s")
+		}
+
+		return sqs.New(sess)
+	}).(*sqs.SQS)
+}
+
+func runSnsSqs(name string, config configInput) {
+	wait.Add(1)
+	go doRunSnsSqs(name, config)
+}
+
+func doRunSnsSqs(name string, configMap configInput) {
+	defer wait.Done()
+	defer log.Printf("%s component of type %s is ready", name, "sns_sqs")
+
+	localConfig := &snsSqsConfig{}
+	unmarshalConfig(configMap, localConfig)
+	snsSqsConfigs[name] = localConfig
+
+	services := "SERVICES=" + strings.Join([]string{
+		"sns",
+		"sqs",
+	}, ",")
+
+	env := []string{services}
+
+	if localConfig.Debug {
+		env = append(env, "DEBUG=1")
+	}
+
+	runContainer("gosoline_test_sns_sqs", ContainerConfig{
```

Comment:
also here i would like to see a const for the container name string

Reply:
✅